pkg/vere/main.c

```diff
@@ -1181,58 +1195,123 @@ _cw_eval_get_input(FILE* fil_u, size_t siz_i)
+    byt_y[i] = fgetc(fil_u);
+  }
+
+  if ( 0x0 != byt_y[0] ) {
+    fprintf(stderr, "corrupted newt passed to cue\n");
+    exit(1);
+  }
+
+  *len_d = (((c3_d)byt_y[1]) <<  0)
+         | (((c3_d)byt_y[2]) <<  8)
+         | (((c3_d)byt_y[3]) << 16)
+         | (((c3_d)byt_y[4]) << 24);
+  byt_y = c3_realloc(byt_y, *len_d);
+
+  for ( i = 0; i < *len_d; ++i ) {
+    byt_y[i] = fgetc(fil_u);
+  }
+
+  return byt_y;
+}
+
 /* _cw_eval(): initialize and run the hoon evaluator
 */
 static void
 _cw_eval(c3_i argc, c3_c* argv[])
 {
-  c3_i ch_i, lid_i;
-  c3_w arg_w;
-  c3_o jam_o = c3n;
-  c3_o kan_o = c3n;
+  u3_mojo std_u;
+  c3_i    ch_i, lid_i;
+  c3_w    arg_w;
+  c3_o    cue_o = c3n;
+  c3_o    jam_o = c3n;
+  c3_o    kan_o = c3n;
+  c3_o    new_o = c3n;
 
   static struct option lop_u[] = {
-    { "loom", required_argument, NULL, c3__loom },
-    { "jam", no_argument, NULL, 'j' },
-    { "jamkhan", no_argument, NULL, 'k' },
+    { "loom", required_argument,  NULL, c3__loom },
+    { "cue",  no_argument,        NULL, 'c'},
+    { "jam",  no_argument,        NULL, 'j' },
+    { "newt", no_argument,        NULL, 'n' },
+    //
     { NULL, 0, NULL, 0 }
   };
 
-  while ( -1 != (ch_i=getopt_long(argc, argv, "", lop_u, &lid_i)) ) {
+  while ( -1 != (ch_i=getopt_long(argc, argv, "cjkn", lop_u, &lid_i)) ) {
     switch ( ch_i ) {
       case c3__loom: {
-        c3_w lom_w;
-        c3_o res_o = _main_readw(optarg, u3a_bits_max+1, &lom_w);
-        if ( (c3n == res_o) || (lom_w < 24) ) {
-          fprintf(stderr, "error: --loom must be >= 24 and <= %zu\r\n", u3a_bits_max);
+        if (_main_readw_loom("loom", &u3_Host.ops_u.lom_y)) {
           exit(1);
         }
-        u3_Host.ops_u.lom_y = lom_w;
+      } break;
+
+      case 'c': {
+        cue_o = c3y;
       } break;
 
       case 'j': {
         jam_o = c3y;
-        kan_o = c3n;
       } break;
 
       case 'k': {
-        jam_o = c3y;
         kan_o = c3y;
       } break;
 
+      case 'n': {
+        new_o = c3y;
+      } break;
+
       case '?': {
         fprintf(stderr, "invalid argument\r\n");
         exit(1);
       } break;
     }
   }
+
+  //  cannot have both jam and cue set
+  //
+  if ( ( c3y == cue_o ) && ( c3y == jam_o ) ) {
+    fprintf(stderr, "cannot enable both jam and cue\r\n");
+    exit(1);
+  }
+  //  newt meaningless without jam or cue set
+  //
+  if ( ( c3y == new_o ) && ( c3n == cue_o ) && ( c3n == jam_o) ) {
+    fprintf(stderr, "newt meaningless w/o jam or cue; ignoring\r\n");
+    new_o = c3n;
+  }
   //  argv[optind] is always "eval"
   //
   if ( optind + 1 != argc ) {
     fprintf(stderr, "invalid command\r\n");
     exit(1);
   }
-  c3_c* evl_c = _cw_eval_get_input(stdin, 10);
+
+  //  configure stdout as u3_mojo
+  //
+  {
+    c3_i err_i;
+    err_i = uv_pipe_init(uv_default_loop(), &std_u.pyp_u, 0);
+    c3_assert(!err_i);
+    uv_pipe_open(&std_u.pyp_u, 1);
+
+    std_u.ptr_v = NULL;
+    std_u.bal_f = _cw_king_fail;
+  }
```

Comment:
This usage of newt and libuv is actually a little odd -- there's no subsequent `uv_run()` invocation, so no async callbacks will ever fire for resource cleanup or error handling. The "newt" protocol is so trivial that "reimplementation" (a la `_cw_eval_get_newt()` here) doesn't bother me -- using the async mechanisms here is cumbersome at best.

But this is not blocking, and will be easy to change if it becomes an actual problem.